src/device/usbd.c

```diff
@@ -1036,4 +1041,60 @@ bool usbd_edpt_stalled(uint8_t rhport, uint8_t ep_addr)
   return _usbd_dev.ep_status[epnum][dir].stalled;
 }
 
+/**
+ * Remove queued xfer complete messages from event queue,
+ * for a particular ep.
+ */
+static void usbd_abort_transfers(uint8_t rhport, uint8_t ep_addr)
```

Comment:
I don't think we need this function for now, let's focus on the dcd_edpt_close() for now, we can address this later on.

Reply:
I think this is needed to flush pending transfers when the edpt is closed. Otherwise, the class driver will receive edpt complete callbacks after the endpoint is closed (and perhaps reopened). The class driver can't know if the transfers happened before or after the endpoint was closed.

Reply:
Yes, you are right, that we need to have some kind of clean up/abort with the currently pending transfer. But I don't really like the way we loop through the task queue. We will need to find another way to address this.

Reply:
please remove this as well.

Reply:
I still couldn't find this event queue filtering is a good approach. It potentially introduce other problems into the stack. In case it does indeed remove an few events. Those are valid and successful transfer, application/class driver e.g MSC could be blocking wait for those. The host close the pipe after those transfer, e.g HOST send eject command to MSC, then close the interface. But since we drop eject transfer, it makes no sense to application.